src/lib/crypto/kimchi_backend/gadgets/xor.ml

```diff
@@ -0,0 +1,333 @@
+open Core_kernel
+
+open Kimchi_backend_common.Plonk_constraint_system.Plonk_constraint
+
+module Bignum_bigint = Snarky_backendless.Backend_extended.Bignum_bigint
+
+(* XOR *)
+
+(* Boolean Xor of length bits *)
+let bxor (type f)
```

Comment:
Do you plan to expose this publicly so circuit designers can use it?

Reply:
i guess, this should be the only function designers should use to build xors inside snarky

Reply:
Since this is an important API, you should probably create an MLI file then to limit what users have access to and document the interface there fully.